src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts

```diff
@@ -616,6 +654,29 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
     this.onSelect(this.selection);
   }
 
+  updateExpanded() {
```

Comment:
Please add some unit tests for this method. 

Also a suggestion how you could refactor this method:
```diff
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts b/src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts
index 8b3178e681..1a30af084d 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts
@@ -655,20 +655,11 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
   }
 
   updateExpanded() {
-    if (typeof this.expanded === 'undefined') {
+    if (_.isUndefined(this.expanded) || this.updateExpandedOnRefresh === 'never') {
       return;
     }
-
-    if (this.updateExpandedOnRefresh === 'never') {
-      return;
-    }
-    let newExpanded: any;
-
-    for (const row of this.data) {
-      if (this.expanded[this.identifier] === row[this.identifier]) {
-        newExpanded = row;
-      }
-    }
+    const expandedId=this.expanded[this.identifier]
+    const newExpanded = _.find(this.data, (row) => expandedId === row[this.identifier])
 
     if (this.updateExpandedOnRefresh === 'onChange' && _.isEqual(this.expanded, newExpanded)) {
       return;
```